src/utils/internalLinks.js

```diff
@@ -5,18 +5,38 @@ import { makeUserId } from '../api/idTypes';
 import type { Narrow, Stream, UserId } from '../types';
 import { topicNarrow, streamNarrow, specialNarrow, pmNarrowFromRecipients } from './narrow';
 import { pmKeyRecipientsFromIds } from './recipient';
+import { ensureUnreachable } from '../generics';
 
-// TODO: Work out what this does, write a jsdoc for its interface, and
-// reimplement using URL object (not just for the realm)
-const getPathsFromUrl = (url: string, realm: URL) => {
-  const paths = url.split(realm.toString()).pop().split('#narrow/').pop()
```

Comment:
With that in mind, if we make some reasonable assumptions [1], then
    the value of
    
      url.split(realm.toString()).pop() // old code
    
    is the same as
    
      url // new code
    
    in the following way: they both share a suffix that starts with
    "#narrow/".

nit: "is the same as" to me means it's in some sense the *same* string, not that it has some relevant subpart in common.

Replacing "is the same as" with "is like", or rewording to "has in common with […] the following: they both […]", would work well.

Reply:
Thanks!

As a side note, I acknowledge that this long commit message doesn't actually prove that the behavior changes are what I say they are. In particular, it doesn't prove that the assumptions in the footnote are the right ones to make.

I don't have any other behavior changes in mind, but I wanted to point this out in case you were waffling about telling me to think harder about this. 🙂

Reply:
> On the other side of the scale: the change doesn't seem like it could be harmful, and we have plenty of other demands on our time.

Yeah, exactly.